src/main/java/org/cryptomator/linux/tray/AppindicatorTrayMenuController.java

```diff
@@ -0,0 +1,123 @@
+package org.cryptomator.linux.tray;
+
+import org.cryptomator.integrations.common.CheckAvailability;
+import org.cryptomator.integrations.common.OperatingSystem;
+import org.cryptomator.integrations.common.Priority;
+import org.cryptomator.integrations.tray.ActionItem;
+import org.cryptomator.integrations.tray.SeparatorItem;
+import org.cryptomator.integrations.tray.SubMenuItem;
+import org.cryptomator.integrations.tray.TrayIconLoader;
+import org.cryptomator.integrations.tray.TrayMenuController;
+import org.cryptomator.integrations.tray.TrayMenuException;
+import org.cryptomator.integrations.tray.TrayMenuItem;
+import org.purejava.appindicator.MemoryAllocator;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.lang.foreign.MemorySegment;
+import java.lang.foreign.SegmentScope;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.nio.file.Paths;
+import java.util.List;
+import java.util.function.Consumer;
+
+import static org.purejava.appindicator.app_indicator_h.*;
+
+@Priority(1000)
+@OperatingSystem(OperatingSystem.Value.LINUX)
+public class AppindicatorTrayMenuController implements TrayMenuController {
+
+	private static final Logger LOG = LoggerFactory.getLogger(AppindicatorTrayMenuController.class);
+
+	private static final SegmentScope SCOPE = SegmentScope.global();
+	private MemorySegment indicator;
+	private MemorySegment menu = gtk_menu_new();
+
+	@CheckAvailability
+	public static boolean isAvailable() {
+		return MemoryAllocator.isLoadedNativeLib();
+	}
+
+	@Override
+	public void showTrayIcon(URI uri, Runnable runnable, String s) throws TrayMenuException {
```

Comment:
I intend to also change the URI to a `Consumer<TrayIconLoader> iconLoader` here.

Reply:
Sure. I did not forget that.